pkg/local-storage/member/controller/volume_snapshot_task_worker_test.go

```diff
@@ -1 +1,494 @@
+		nodeTaskQueue               *common.TaskQueue
+		k8sNodeTaskQueue            *common.TaskQueue
+		volumeTaskQueue             *common.TaskQueue
+		volumeExpandTaskQueue       *common.TaskQueue
+		volumeMigrateTaskQueue      *common.TaskQueue
+		volumeGroupMigrateTaskQueue *common.TaskQueue
+		volumeConvertTaskQueue      *common.TaskQueue
+		volumeGroupConvertTaskQueue *common.TaskQueue
+		volumeSnapshotTaskQueue     *common.TaskQueue
+		localNodes                  map[string]v1alpha1.State
+		logger                      *log.Entry
+		lock                        sync.Mutex
+	}
+	type args struct {
+		snapShot *v1alpha1.LocalVolumeSnapshot
+	}
+	client, _ := CreateFakeClient()
+
+	// Create LocalVolume
+	lv := GenFakeLocalVolumeObject()
+	lv.Name = fakeLocalVolumeName
+	lv.Spec.VolumeGroup = fakeLocalVolumeGroupName
+	lv.Spec.PersistentVolumeClaimName = fakePersistentVolumeClaimName
+	lv.Spec.ReplicaNumber = 1
+	lv.Spec.Convertible = true
+	lv.Spec.PersistentVolumeClaimNamespace = fakeNamespace
+	err := client.Create(context.Background(), lv)
+	if err != nil {
+		t.Errorf("Create LocalVolume fail %v", err)
+	}
+
+	// Create LocalVolumeGroup
+	lvg := GenFakeLocalVolumeGroupObject()
+	lvg.Name = fakeLocalVolumeGroupName
+	lvg.Namespace = fakeNamespace
+	err = client.Create(context.Background(), lvg)
+	if err != nil {
+		t.Errorf("Create LocalVolumeGroup fail %v", err)
+	}
+
+	// Create LocalVolumeConvert
+	lvs := GenFakeLocalVolumeSnapShotObject()
+	lvs.Namespace = fakeNamespace
+
+	err = client.Create(context.Background(), lvs)
+	if err != nil {
+		t.Errorf("Create VolumeSnapshot fail %v", err)
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		// TODO: Add test cases.
+		{
+			args: args{
+				snapShot: lvs,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := &manager{
+				name:                        fakeManagerName,
+				namespace:                   fakeNamespace,
+				apiClient:                   client,
+				nodeTaskQueue:               common.NewTaskQueue("NodeTask", maxRetries),
+				k8sNodeTaskQueue:            common.NewTaskQueue("K8sNodeTask", maxRetries),
+				volumeTaskQueue:             common.NewTaskQueue("VolumeTask", maxRetries),
+				volumeExpandTaskQueue:       common.NewTaskQueue("VolumeExpandTask", maxRetries),
+				volumeMigrateTaskQueue:      common.NewTaskQueue("VolumeMigrateTask", maxRetries),
+				volumeGroupMigrateTaskQueue: common.NewTaskQueue("VolumeGroupMigrateTask", maxRetries),
+				volumeConvertTaskQueue:      common.NewTaskQueue("VolumeConvertTask", maxRetries),
+				volumeGroupConvertTaskQueue: common.NewTaskQueue("VolumeGroupConvertTask", maxRetries),
+				localNodes:                  map[string]v1alpha1.State{},
+				logger:                      log.WithField("Module", "ControllerManager"),
+			}
+			if err := m.volumeSnapshotSubmit(tt.args.snapShot); (err != nil) != tt.wantErr {
```

Comment:
expect the checking for the return value of error, but also should check the wanted status for this snapshot after reconcile?